keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -0,0 +1,578 @@
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Assumptions](#assumptions)
+  - [Configuration](#configuration)
+    - [Interoperability](#interoperability)
+    - [Feature Gate](#feature-gate)
+  - [API](#api)
+    - [Future API Expansion](#future-api-expansion)
+  - [Kube-Proxy](#kube-proxy)
+  - [EndpointSlice Controller](#endpointslice-controller)
+    - [Example](#example)
+    - [Overload](#overload)
+    - [Handling Node Updates](#handling-node-updates)
+  - [Future Expansion](#future-expansion)
+  - [Test Plan](#test-plan)
+    - [Controller Unit Tests](#controller-unit-tests)
+    - [Kube-Proxy Unit Tests](#kube-proxy-unit-tests)
+  - [Observability](#observability)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [x] (R) Graduation criteria is in place
+- [x] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes clusters are increasingly deployed in multi-zone environments but
+unfortunately network routing has not caught up with that. This KEP proposes an
+automatic topology aware hinting mechanism with a new Service annotation. This
+new annotation will allow EndpointSlice producers to provide hints to
```

Comment:
This says annotation, but I think you settled on a field?

Reply:
Yeah this is way more confusing than I intended. When I wrote this I was trying to describe the annotation that would be used to enable this on each service. Will update this to be more clear.

Reply:
Cleaned this up and removed the reference to the annotation entirely from this section - feels like coverage in design details is sufficient for that.